unix/vncserver/selinux/vncsession.te

```diff
@@ -34,12 +34,10 @@ allow vnc_session_t self:capability { chown dac_override dac_read_search fowner
 allow vnc_session_t self:process { getcap setexec setrlimit setsched };
 allow vnc_session_t self:fifo_file rw_fifo_file_perms;
 
+# Allow access to /proc/sys/fs/nr_open
+# Needed when the nofile limit is set to unlimited.
 optional_policy(`
-        gen_require(`
-                type sysctl_fs_t;
-        ')
-        allow vnc_session_t sysctl_fs_t:dir search;
-        allow vnc_session_t sysctl_fs_t:file { getattr open read };
+    kernel_read_fs_sysctls(vnc_session_t)
```

Comment:
kernel_* should not appear in an optional block, it should be moved too (current) line 46